### PR TITLE
Introduce telemetry configuration repository

### PR DESCRIPTION
**Pull request by ankithkonda**
We've changed the way how Octopus Deploy collects and sends telemetry, for better transparency and visibility.

This PR introduces a telemetry configuration repository to easily enable/disable telemetry

### CODE REVIEW

File: source/Octopus.Server.Client/Model/TelemetryConfigurationResource.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using Octopus.Client.Extensibility.Attributes;
+
+namespace Octopus.Client.Model
+{
+    public class TelemetryConfigurationResource: Resource
+    {
+        [Writeable]
+        public bool Enabled { get; set; }
+
+        public DateTimeOffset ShowAsNewUntil { get; set; }
```
Comment (tothegills) [inline source/Octopus.Server.Client/Model/TelemetryConfigurationResource.cs:11]: Do we want `ShowAsNewUntil` as part of the client api?

File: source/Octopus.Server.Client/Repositories/TelemetryConfigurationRepository.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        public TelemetryConfigurationResource ModifyTelemetryConfiguration(TelemetryConfigurationResource resource)
+        {
+            return repository.Client.Update(repository.Link(LinkName), resource);
+        }
+
+        public TelemetryConfigurationResource EnableTelemetry()
+        {
+            return repository.Client.Update(repository.Link(LinkName), new TelemetryConfigurationResource
```
Comment (tothegills) [inline source/Octopus.Server.Client/Repositories/TelemetryConfigurationRepository.cs:36]: Use `ModifyTelemetryConfiguration`?

File: source/Octopus.Server.Client/Repositories/TelemetryConfigurationRepository.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            {
+                Enabled = true
+            });
+        }
+
+        public TelemetryConfigurationResource DisableTelemetry()
+        {
+            return repository.Client.Update(repository.Link(LinkName), new TelemetryConfigurationResource
```
Comment (tothegills) [inline source/Octopus.Server.Client/Repositories/TelemetryConfigurationRepository.cs:44]: Use `ModifyTelemetryConfiguration`?

**Review by tothegills — COMMENTED** [review]
Not sure about including the cutoff time, it seems like something we are interested in for the sake of our portal.

**Review by tothegills — APPROVED** [review]

**Comment by ankithkonda** [comment]
I've had to mark the macos test as successful manually there's a known in with the mac agent with SSL https://octopusdeploy.slack.com/archives/C012VMX1YMQ/p1636980261180600

**Pull request merged**